packages/grid/src/GridMetricCalculator.ts

```diff
@@ -37,44 +82,73 @@ class GridMetricCalculator {
   /**
    * Get the coordinates of floating items in one dimension.
    * Can be used for getting the y coordinates of floating rows, or x coordinates of floating columns, calculated using the `sizeMap` passed in.
-   * @param {number} startCount The number of floating items at the start (ie. `floatingTopRowCount` for rows, `floatingLeftColumnCount` for columns)
-   * @param {number} endCount The number of floating items at the end (ie. `floatingBottomRowCount` for rows, `floatingRightColumnCount` for columns)
-   * @param {number} totalCount Total number of items in this dimension (ie. `rowCount` for rows, `columnCount` for columns)
-   * @param {number} max The max coordinate value (ie. `maxY` for rows, `maxX` for columns)
-   * @param {Map<number, number>} sizeMap Map from index to size of item (ie. `rowHeightMap` for rows, `columnWidthMap` for columns)
+   * @param startCount The number of floating items at the start (ie. `floatingTopRowCount` for rows, `floatingLeftColumnCount` for columns)
+   * @param endCount The number of floating items at the end (ie. `floatingBottomRowCount` for rows, `floatingRightColumnCount` for columns)
+   * @param totalCount Total number of items in this dimension (ie. `rowCount` for rows, `columnCount` for columns)
+   * @param max The max coordinate value (ie. `maxY` for rows, `maxX` for columns)
+   * @param sizeMap Map from index to size of item (ie. `rowHeightMap` for rows, `columnWidthMap` for columns)
    */
   static getFloatingCoordinates(
-    startCount,
-    endCount,
-    totalCount,
-    max,
-    sizeMap
-  ) {
+    startCount: number,
+    endCount: number,
+    totalCount: number,
+    max: number,
+    sizeMap: SizeMap
+  ): CoordinateMap {
     const coordinates = new Map();
     let x = 0;
     for (let i = 0; i < startCount && i < totalCount; i += 1) {
       coordinates.set(i, x);
-      x += sizeMap.get(i);
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      x += sizeMap.get(i)!;
     }
 
     x = max;
     for (let i = 0; i < endCount && totalCount - i - 1 >= 0; i += 1) {
-      x -= sizeMap.get(totalCount - i - 1);
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      x -= sizeMap.get(totalCount - i - 1)!;
```

Comment:
There's a few places where as a result of converting to TS this started getting flagged. I did add `?? 0` in most places, but really - it _should_ be an error, since if that value doesn't exist in the map, then the grid metric calculations were incorrect, therefore this calculation is going to be invalid.

In any case, I was incorrectly using the non-null assertion here as you pointed out. I expected it to throw, similar to in Swift with forced unwrapping. But it appears this is only a compile time flag, and at runtime it just returns `undefined`. So `?? 0` it is